lib/logstash/outputs/elasticsearch.rb

```diff
@@ -50,114 +49,18 @@
 # Keep in mind that a connection with keepalive enabled will
 # not reevaluate its DNS value while the keepalive is in effect.
 class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
-  attr_reader :client
+  require "logstash/outputs/elasticsearch/http_client"
+  require "logstash/outputs/elasticsearch/http_client_builder"
+  require "logstash/outputs/elasticsearch/common_configs"
+  require "logstash/outputs/elasticsearch/common"
 
-  include Stud::Buffer
-  RETRYABLE_CODES = [409, 429, 503]
-  SUCCESS_CODES = [200, 201]
+  # Most config options are defined here!
+  include(LogStash::Outputs::ElasticSearch::CommonConfigs)
```

Comment:
Tell me about what differentiates config settings that go into CommonConfigs vs ones that are declared in this file?